templates/site/js/main.js

```diff
@@ -114,129 +49,138 @@ function login(){
-    return fetch(githubURL).then(response => {
-        return response.json().then(json => {
+function getUserDetails(){
+    const loggedUserURLFragment = "/magpie/users/current";
+    return fetch(loggedUserURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
+            }
+        }).then(response => {return response.json()}).then(json =>{
             return json;
         })
-    })
 }
-function getBaseURL(json){
-    return json.then( sessionData =>
-    {
-        var urlArray = sessionData.url.split("/magpie");
-        return urlArray[0];
+
+function updateUserDetails(){
+    var username;
+    var sessionEmail;
+
+    getUserDetails().then(json => {
+        username = json.user["user_name"];
+        sessionEmail = json.user["email"];
+        var password = document.getElementById("userPassword").value;
+        var email = document.getElementById("settingsEditEmail").value;
+
+        showSpinner()
+
+        if (password != "" || password != null){
+            updateUserPassword(username, password);
+        }
+
+        if(email != sessionEmail && email != null){
+            updateUserEmail(username, email);
+        }
     })
+
 }
 
-function getSessionDetails(){
-    //TODO Remove sessionURL and use sessionURLFragment
-    //TODO Remove nodeURL and use {MAGPIE_URL}
-    const sessionURLFragment = "session";
-    const sessionURL = nodeURL + "/magpie/session"
-    //const testSessionURL = testNodeURL + "/magpie/session"
-    return fetch(sessionURL,{
-        method: "GET",
+function updateUserEmail(username,email){
+    const updateURLFragment = "/magpie/users/" + username;
+
+    fetch(updateURLFragment, {
+        method: "PATCH",
         headers: {
             "Accept": "application/json, text/plain",
             "Content-Type": "application/json"
+        },
+        body:JSON.stringify({
+            "email": email
+        })
+    }).then(response => response.json()).then(json =>{
+            if(json.code && json.code == 200){
+                showSaveCheckmark()
             }
-        }).then(response => { return response.json().then(json =>{
-                return json;
-            })
         })
 }
 
-//User functions
-function getUserDetails(username, cookie){
-    //Use fetch to get user details from user/username endpoint in api
-    //Use details returned to populate fields
-
-    //TODO: When done remove loggedUserUrl and put loggedUserURLFragment for the fetch url.  base url should resolve when run from birdhouse
-    const loggedUserURL = nodeURL + "/magpie/users/" + username;
-    //const loggedUserURL = nodeURL + "/magpie/users/current";
-    //const loggedUserURLFragment = "users/current";
-    return fetch(loggedUserURL,{
-        method: "GET",
+function updateUserPassword(username, password){
+    const updateURLFragment = "/magpie/users/" + username;
```

Comment:
It isn't necessary to get the current user here. Just use the `/magpie/users/current` path

Reply:
Removed this function.